src/App.jsx

```diff
@@ -14,45 +15,99 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_ALPHABET = 'Sort alphabetically';
+const SORT_LENGTH = 'Sort by length';
+
+export const App = () => {
+  const [visibleGoods, setVisibleGoods] = useState(goodsFromServer);
+  const [sortField, setSortField] = useState(null);
+  const [reverse, setReverse] = useState(false);
+
+  const sortByAlphabet = () => {
+    if (reverse) {
+      setVisibleGoods(
+        [...visibleGoods].sort().reverse(),
+      );
+    } else {
+      setVisibleGoods(
+        [...visibleGoods].sort(),
+      );
+    }
+
+    setSortField(SORT_ALPHABET);
+  };
+
+  const sortByLength = () => {
+    if (reverse) {
+      setVisibleGoods(
+        [...visibleGoods].sort((good1, good2) => good2.length - good1.length),
+      );
+    } else {
+      setVisibleGoods(
+        [...visibleGoods].sort((good1, good2) => good1.length - good2.length),
+      );
+    }
+
+    setSortField(SORT_LENGTH);
+  };
+
+  const reverseGood = () => {
+    const reversedGoods = [...visibleGoods].reverse();
+
```

Comment:
better to create the one function that will take goods, sortField, and reverse as the parameters and sort the array there